app/src/main/kotlin/com/intive/picover/common/viewmodel/MVIViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.intive.picover.common.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.common.viewmodel.event.Event
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+abstract class MVIViewModel<T : Event> : ViewModel() {
+
+	private val event: MutableSharedFlow<T> = MutableSharedFlow()
+
+	init {
+		subscribeToEvents()
+	}
+
+	open fun handleEvents(event: T) {}
```

Comment:
minor: the name says about events but the param is a single event 😄

Reply:
changed :) thanks !